File: src/ide/webstorm/src/main/java/com/point/i18n/webstorm/services/StorageService.java
package com.point.i18n.webstorm.services;

import com.intellij.ide.util.PropertiesComponent;

/**
 * Service for storing tokens and settings.
 * Java implementation using PropertiesComponent.
 */
public class StorageService {
    private static final String ACCESS_TOKEN_KEY = "point.i18n.accessToken";
    private static final String REFRESH_TOKEN_KEY = "point.i18n.refreshToken";
    private static final String REFRESH_EXPIRES_IN_KEY = "point.i18n.refreshExpiresIn";
    
    private final PropertiesComponent propertiesComponent;
    
    public StorageService() {
        this.propertiesComponent = PropertiesComponent.getInstance();
    }
    
    public String getAccessToken() {
        return propertiesComponent.getValue(ACCESS_TOKEN_KEY);
    }
    
    public String getRefreshToken() {
        return propertiesComponent.getValue(REFRESH_TOKEN_KEY);
    }
    
    public Integer getRefreshExpiresIn() {
        String value = propertiesComponent.getValue(REFRESH_EXPIRES_IN_KEY);
        return value != null ? Integer.parseInt(value) : null;
    }
    
    public void saveTokens(ApiService.AuthResponse authResponse) {
        propertiesComponent.setValue(ACCESS_TOKEN_KEY, authResponse.access_token);
        propertiesComponent.setValue(REFRESH_TOKEN_KEY, authResponse.refresh_token);
        propertiesComponent.setValue(REFRESH_EXPIRES_IN_KEY, String.valueOf(authResponse.refresh_expires_in));
    }
    
    public void deleteTokens() {
        propertiesComponent.unsetValue(ACCESS_TOKEN_KEY);
        propertiesComponent.unsetValue(REFRESH_TOKEN_KEY);
        propertiesComponent.unsetValue(REFRESH_EXPIRES_IN_KEY);
    }
}

File: src/ide/webstorm/services/storage.service.ts
import { IStorageService, TokenData } from '../../../shared/types';

// IntelliJ Platform API types (упрощенные)
declare const com: any;
declare const java: any;

export class WebStormStorageService implements IStorageService {
    private readonly ACCESS_TOKEN_KEY = 'i18nRemote.accessToken';
    private readonly REFRESH_TOKEN_KEY = 'i18nRemote.refreshToken';
    private readonly REFRESH_EXPIRES_IN_KEY = 'i18nRemote.refreshExpiresIn';

    private getProperties(): any {
        // Используем PropertiesComponent для хранения данных
        // @ts-ignore
        return com.intellij.ide.util.PropertiesComponent.getInstance();
    }

    async getAccessToken(): Promise<string | undefined> {
        try {
            const props = this.getProperties();
            const value = props.getValue(this.ACCESS_TOKEN_KEY);
            return value || undefined;
        } catch (error) {
            console.error('Error getting access token:', error);
            return undefined;
        }
    }

    async getRefreshToken(): Promise<string | undefined> {
        try {
            const props = this.getProperties();
            const value = props.getValue(this.REFRESH_TOKEN_KEY);
            return value || undefined;
        } catch (error) {
            console.error('Error getting refresh token:', error);
            return undefined;
        }
    }

    async getRefreshExpiresIn(): Promise<number | undefined> {
        try {
            const props = this.getProperties();
            const value = props.getValue(this.REFRESH_EXPIRES_IN_KEY);
            return value ? parseInt(value, 10) : undefined;
        } catch (error) {
            console.error('Error getting refresh expires in:', error);
            return undefined;
        }
    }

    async getToken(): Promise<string | undefined> {
        return await this.getAccessToken();
    }

    async saveTokens(authResponse: TokenData): Promise<void> {
        try {
            const props = this.getProperties();
            props.setValue(this.ACCESS_TOKEN_KEY, authResponse.access_token);
            props.setValue(this.REFRESH_TOKEN_KEY, authResponse.refresh_token);
            props.setValue(this.REFRESH_EXPIRES_IN_KEY, authResponse.refresh_expires_in.toString());
        } catch (error) {
            console.error('Error saving tokens:', error);
            throw error;
        }
    }

    async saveToken(token: string): Promise<void> {
        try {
            const props = this.getProperties();
            props.setValue(this.ACCESS_TOKEN_KEY, token);
        } catch (error) {
            console.error('Error saving token:', error);
            throw error;
        }
    }

    async deleteTokens(): Promise<void> {
        try {
            const props = this.getProperties();
            props.unsetValue(this.ACCESS_TOKEN_KEY);
            props.unsetValue(this.REFRESH_TOKEN_KEY);
            props.unsetValue(this.REFRESH_EXPIRES_IN_KEY);
        } catch (error) {
            console.error('Error deleting tokens:', error);
            throw error;
        }
    }

    async deleteToken(): Promise<void> {
        await this.deleteTokens();
    }
}
